fideslog/api/models/analytics_event.py

```diff
@@ -81,12 +93,25 @@ def check_not_an_email_address(cls, value: str) -> str:
         return value
 
     @validator("endpoint")
-    def check_no_hostname(cls, value: str) -> str:
+    def validate_endpoint_format(cls, value: str) -> str:
         """
-        Ensure that endpoints contain only the URL path.
+        Ensure that endpoint contains method and path, and that path component contains only the URL path.
         """
 
-        return urlparse(value).path
+        endpoint_components = value.split(":")
+        assert (
+            len(endpoint_components) == 2
+        ), "endpoint must contain only the HTTP method and URL path, delimited by a colon"
+        endpoint_components[0] = endpoint_components[0].strip().upper()
+        assert (
+            endpoint_components[0].strip() in ALLOWED_HTTP_METHODS
+        ), f"HTTP method must be one of {', '.join(ALLOWED_HTTP_METHODS)}"
+        assert (
+            endpoint_components[1].strip()
+            == urlparse(endpoint_components[1].strip()).path
+        ), "endpoint must contain only the URL path"
+        value_with_uppercase_method = ":".join(endpoint_components)
+        return value_with_uppercase_method
```

Comment:
You've done the work to up-case the HTTP method on L105, but you're still returning whatever the original value was here.
```suggestion

        return f"{endpoint_components[0]}: {endpoint_components[1].strip()}"
```
Separately, with this change, we'd be calling `endpoint_components[1].strip()` for a third time. Maybe add another line below L105 like:
```python
path = endpoint_components[1].strip()
```
and if you do this, maybe also just create a new variable on L105 instead of overwriting `endpoint_components[0]`? I think it would be more clear that way.